sql/core/src/test/scala/org/apache/spark/sql/streaming/FileStreamSinkSuite.scala

```diff
@@ -512,3 +429,174 @@ class FileStreamSinkSuite extends StreamTest {
     }
   }
 }
+
+class FileStreamSinkV1Suite extends FileStreamSinkSuite {
+  import testImplicits._
+
+  override protected def sparkConf: SparkConf =
+    super
+      .sparkConf
+      .set(SQLConf.USE_V1_SOURCE_READER_LIST, "csv,json,orc,text,parquet")
+      .set(SQLConf.USE_V1_SOURCE_WRITER_LIST, "csv,json,orc,text,parquet")
+
+  test("partitioned writing and batch reading") {
```

Comment:
can you highlight the difference between this test case and the one in `FileStreamSinkV2Suite`?

Reply:
For V1 suite, it uses Parquet V1, and it matches `HadoopFsRelation` / `FileScanRDD` to check if the plan is as expected. Also, partition pruning is tested.

For V2 suite, it uses Parquet V2, and it matches  `FileTable` / `BatchScanExec`  to check if the plan is as expected. As partition pruning is not supported in V2 yet, we can't test it.

I can abstract the code if the two cases look duplicated to you.

Reply:
Or, do you mean changing the test case name?

Reply:
let's abstract the code, this test case is too long to duplicate.